src/grcov/covdir.rs

```diff
@@ -107,24 +110,30 @@ impl Covdir {
 }
 
 #[inline]
-fn get_directory(covdir_json: &Value) -> Result<String> {
-    Ok(covdir_json
-        .get("name")
-        .ok_or(Error::Conversion)?
-        .as_str()
-        .ok_or(Error::Conversion)?
-        .chars()
-        .filter(|c| !c.is_whitespace())
-        .collect::<String>())
+fn get_directory(covdir_json: &Value) -> Result<PathBuf> {
+    Ok(PathBuf::from(
+        covdir_json
+            .get("name")
+            .ok_or(Error::Conversion)?
+            .as_str()
+            .ok_or(Error::Conversion)?,
+    ))
+}
+
+#[inline]
+fn get_file_path(project_path: &Path, file_relative_path: &Path) -> PathBuf {
+    let file_path = project_path.to_path_buf().join(file_relative_path);
+
+    PathBuf::from(file_path.to_string_lossy().replace('\\', "/"))
 }
 
 #[inline]
-fn parse_json_coverage(json_coverage: &[Value]) -> Vec<i32> {
+fn parse_coverage(json_coverage: &[Value]) -> Vec<Option<i32>> {
     json_coverage
         .iter()
         .filter_map(|c| c.as_i64())
-        .map(|v| v as i32)
-        .collect::<Vec<i32>>()
+        .map(|v| if v == -1 { None } else { Some(v as i32) })
```

Comment:
The order is important because each element represents a line in the file. I mapped `-1` to `Option` to have consistency with `coveralls.json` variant (which uses `null` values) and therefore have a single format.

Reply:
Ok, perfect! Can we write a comment to explain that?